### PR TITLE
Copy bulk/shear modulus methods to ElasticityTensorTools

### PR DESCRIPTION
**Pull request by bwspenc**
ref #9215

This is the first step.  After I have removed all usages of the one
in RadialReturnStressUpdate in apps, I will remove these from that
class.

### CODE REVIEW

File: modules/tensor_mechanics/src/utils/ElasticityTensorTools.C [hunk 1]
```diff
@@ -108,4 +108,26 @@ momentJacobianWC(const RankFourTensor & r4t, unsigned int i, unsigned int k, Rea
 
   return test * phi * sum;
 }
+
+Real
+getIsotropicShearModulus(const RankFourTensor & elasticity_tensor)
+{
+  const Real shear_modulus = elasticity_tensor(0, 1, 0, 1);
+  if (shear_modulus != elasticity_tensor(0, 2, 0, 2))
+    mooseError("Check to ensure that your Elasticity Tensor is truly Isotropic");
```
Comment (dschwen) [inline modules/tensor_mechanics/src/utils/ElasticityTensorTools.C]: The correct isotropy check is much more complicated and it is implemented in R4T. Do we want to keep this hack here?

Comment (bwspenc) [inline modules/tensor_mechanics/src/utils/ElasticityTensorTools.C]: Maybe it is best to just remove the check here entirely and assume that anyone using this has already done the more expensive check.

**Comment by bwspenc** [comment]
@WilkAndy Would you mind taking a look at the changes I made to your models?  I'm adding a method that all classes that derive from StressUpdateBase must overwrite that is used to see whether the model requires the elasticity tensor to be isotropic.

The RadialReturnStressUpdate models require isotropic tensor, but I'm not sure about your models.  The Cosserat models definitely do not require an isotropic tensor, but I'm not sure about the DruckerPrager and WeakPlane models.

The reason for adding this check is that I'm removing the check from the function that these models call to get the shear modulus, so it should be checked somewhere else.  Your models are not calling that function, but they could.  It looks like they require other characteristics of the tensor, and it might be good to add similar checks for those.

File: modules/tensor_mechanics/include/materials/CappedWeakInclinedPlaneStressUpdate.h [hunk 2]
```diff
@@ -30,6 +30,11 @@ class CappedWeakInclinedPlaneStressUpdate : public CappedWeakPlaneStressUpdate
 public:
   CappedWeakInclinedPlaneStressUpdate(const InputParameters & parameters);
 
+  /**
+   * Does the model require the elasticity tensor to be isotropic?
+   */
+  bool requiresIsotropicTensor() override { return true; }
```
Comment (WilkAndy) [inline modules/tensor_mechanics/include/materials/CappedWeakInclinedPlaneStressUpdate.h]: should be ```false```

File: modules/tensor_mechanics/include/materials/CappedWeakPlaneStressUpdate.h [hunk 3]
```diff
@@ -31,6 +31,11 @@ class CappedWeakPlaneStressUpdate : public TwoParameterPlasticityStressUpdate
 public:
   CappedWeakPlaneStressUpdate(const InputParameters & parameters);
 
+  /**
+   * Does the model require the elasticity tensor to be isotropic?
+   */
+  bool requiresIsotropicTensor() override { return true; }
```
Comment (WilkAndy) [inline modules/tensor_mechanics/include/materials/CappedWeakPlaneStressUpdate.h]: should be ```false```

**Review by WilkAndy — CHANGES_REQUESTED** [review]
Thanks @bwspenc

**Comment by bwspenc** [comment]
Thanks for taking a look, @WilkAndy.  I fixed those two models.

**Comment by bwspenc** [comment]
@dschwen I addressed @WilkAndy's comment, and I think this is ready to go.

**Review by dschwen — APPROVED** [review]

**Pull request merged**